src/isar/state_machine/state_machine.py

```diff
@@ -480,8 +518,8 @@ def publish_task_status(self) -> None:
             retain=False,
         )
 
-    def publish_step_status(self) -> None:
-        """Publishes the current step status to the MQTT Broker"""
+    def publish_step_status(self, step: Step) -> None:
```

Comment:
Should step have a default value so that it can be an optional parameter?

Reply:
Same reason as for task.